python/src/scipp/plot/profiler.py

```diff
@@ -0,0 +1,340 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @author Neil Vaytet
+
+# Scipp imports
+from .plot import plot
+from .render import render_plot
+from .plot_2d import Slicer2d
+from ..show import _hex_to_rgb
+from .tools import to_bin_edges
+from .._utils import name_with_unit
+from .._scipp import core as sc
+from .. import detail
+
+# Other imports
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
+from matplotlib.collections import PathCollection
+
+
+def profiler(scipp_obj_dict=None,
+             axes=None,
+             masks=None,
+             filename=None,
+             figsize=None,
+             ax=None,
+             cax=None,
+             aspect=None,
+             cmap=None,
+             log=False,
+             vmin=None,
+             vmax=None,
+             color=None,
+             logx=False,
+             logy=False,
+             logxy=False,
+             resolution=None):
+    """
+    Plot a 2D slice through a N dimensional dataset. For every dimension above
+    2, a slider is created to adjust the position of the slice in that
+    particular dimension.
+    """
+
+    sv = Profiler(scipp_obj_dict=scipp_obj_dict,
+                  axes=axes,
+                  masks=masks,
+                  ax=ax,
+                  cax=cax,
+                  aspect=aspect,
+                  cmap=cmap,
+                  log=log,
+                  vmin=vmin,
+                  vmax=vmax,
+                  color=color,
+                  logx=logx or logxy,
+                  logy=logy or logxy,
+                  resolution=resolution)
+
+    if ax is None:
+        render_plot(figure=sv.fig, widgets=sv.vbox, filename=filename)
+
+    return sv
+
+
+class Profiler(Slicer2d):
+    def __init__(self,
+                 scipp_obj_dict=None,
+                 axes=None,
+                 masks=None,
+                 ax=None,
+                 cax=None,
+                 aspect=None,
+                 cmap=None,
+                 log=None,
+                 vmin=None,
+                 vmax=None,
+                 color=None,
+                 logx=False,
+                 logy=False,
+                 resolution=None):
+
+        # Variables for profile plotting from pick need to be set before
+        # calling the init from Slicer2d as it calls update_axes() and
+        # update_slice(), which require these variables to be defined.
+        self.profile_viewer = None
+        self.profile_key = None
+        self.slice_pos_rectangle = None
+        self.profile_scatter = None
+        self.profile_update_lock = False
+        self.ax_pick = None
+        self.log = log
```

Comment:
Prefix "private" variables and methods with `_`, as recommended by pep8?

Reply:
I would suggest doing this across all the plotting code as a major cleanup job